content/blog/2016/2016-10-11-stage-lock-milestone.adoc

```diff
@@ -0,0 +1,165 @@
+
+In order to improve upon each of these areas independently we decided to break this functionality into discrete steps rather than push more and more features into an already packed `stage` step.
+
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Stage+Step+Plugin[stage] - the `stage` step remains but is now focused on grouping steps and providing boundaries for Pipeline segments.
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Lockable+Resources+Plugin[lock] - the `lock` step throttles the number concurrent builds in a defined section of the Pipeline.
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Milestone+Step+Plugin[milestone] - the `milestone` step automatically discards builds that will finish out of order and become stale.
+
+Separating these concerns into explicit, independent steps allows for much greater control of Pipelines and broadens the set of possible use cases.
+
+=== Stage
+
+The `stage` step is a primary building block in Pipeline, dividing the steps of a Pipeline into explicit units and helping to visualize the progress. The visualization plugin is called "Stage View" due to this function of `stage`.
+
+Beginning with version 2.2 of `Pipeline Stage Step` plugin, the `stage` step now requires a block argument, wrapping all steps within the defined stage. This makes the boundaries of where each `stage` begins and ends obvious and predictable.
+
+In addition, the concurrency argument of `stage` has now been removed to make this step more concise; responsibility for concurrency control has been delegated to the `lock` step.
+
+[source, groovy]
+----
+stage("Build"){
+  doSomething()
+  sh 'echo $PATH'
+}
+----
+
+Omitting the block from `stage` and using the concurrency argument are now deprecated in Pipeline. Pipelines using this syntax will continue to function but will produce a warning in the console log:
+----
+Using the 'stage' step without a block argument is deprecated
+----
+This message is only a reminder to update your Pipeline scripts; none of your Pipelines will stop working. If we reach a point where the old syntax is to be removed we will make an announcement prior to the change. We do, however, recommend that you update your existing Pipelines to utilize the new syntax.
+
+*note:* Stage View and Blue Ocean will both work with either the old `stage` syntax or the new.
+
+=== Lock
+
+Rather than attempt to limit the number of concurrent builds of a job using the `stage`, we now rely on the `Lockable Resources` plugin and the `lock` step to control this. The `lock` step is limited to a concurrency of one (1) build but it provides much greater flexibility in designating where jobs should be restricted.
```

Comment:
Those should be regular double quotes around "Lockable Resources", rather than `code formatting quotes`.